web-console/src/components/filler.tsx

```diff
@@ -257,3 +259,63 @@ export class TagInput extends React.Component<TagInputProps, { stringValue: stri
     />;
   }
 }
+
+interface JSONInputProps extends React.Props<any>{
+  onChange: (newValue: string) => void;
+  value: JSON;
+  updateErrors: ((newValue: string) => void)
+}
+
+interface JSONInputState{
+  stringValue: string;
+}
+
+export class JSONInput extends React.Component<JSONInputProps, JSONInputState> {
+  constructor(props: JSONInputProps) {
+    super(props);
+    this.state = {
+      stringValue: ""
+    }
+  }
+
+  componentDidMount(): void {
+    const { value } = this.props;
+    const stringValue = parseJSONToString(value);
+    this.setState({
+      stringValue: stringValue
+    })
+  }
```

Comment:
you need to also implement a `getDerivedStateFromProps` because if the props change the state (stringValue) will not currently be updated

Reply:
Currently JSON input works in this way: Take in JSON from auto-form through props as value, convert it to a string, and will only update the JSON data in auto-form when the string value entered is valid. If we use `getDerivedStateFromProps`, does this mean we will constantly check if value from props matches value from state, which may not work as expected, since we want to update the string value in state however we want, but only update the JSON value when that string value is valid?

Reply:
You are correct, but what the current code does not address is that if the props change underneath outside the autoform it should be reflected

Reply:
Would it be a better idea to use `componentDidUpdate` or `componentWillReceiveProps` (though deprecated) than `getDerivedStateFromProps`? Since we are mainly checking if the previous and current props are matched when props are changed outside the autoform?